### DISCUSSION TITLE
Pseudocodigo de BANANER Etapa 0

### DISCUSSION THREAD

**Discussion by ypne** [IIC2413/Syllabus-2024-2#47] [General]
El pseudocódigo  de bananer etapa0

FUNCTION Insert (at1, ..atn) RETURN INT { % Carga los atributos  at1 .. atn en las matrices correspondientes int retorna 1 éxito 2 error
% supongamos que los at1..atk pertenecen a la matriz 1 y los restantes a la matriz 2

%llenado de la matriz1 sin duplicados
        FOR i IN 1..N   {   % N es el tamaño máximo de la matriz
                IF Matriz1 [i] = at1..atk RETURN 0  %el registro ya está en la matriz, se descarta y retorna éxito
        IF Matriz[k] := at1..atk RETURN 1 %se asignan los atributos al primer espacio vacío de la matriz (k) y retorna éxito
       ELSE RETURN 1 retorna fracaso en insertar en la matriz 1

%llenado de la matriz2 sin duplicados
        FOR i IN 1..M       % M es el tamaño máximo de la matriz
                IF Matriz2 [i] = atk+1..atn RETURN 0  %el registro ya está en la matriz, se descarta y retorna éxito
        IF Matriz[j] := at1..atk RETURN TRUE %se asignan los atributos al primer espacio vacío de la matriz (j) y retorna éxito
       ELSE RETURN 2 retorna fracaso en insertar en la matriz 2
}

     
FUNCTION CARGA ($1) Boolean
  i=1
 resultado=TRUE
  WHILE NOT EOF ($1)  % mientras no se acabe el archivo $1 pasado como parámetro
         IF READLN ($1, at1..atn).   % lee correctamente una linea del archivo $1 y lo carga en las variables at1 a atn
               CASE INSERT(at1..atn)  
                  0 CONTINUE %la inserción fue correcta, "happy path"
                  1 PRINTF "error de inserción en matriz 1 $at1..$atk resultado=FALSE
                  2 PRINTF "error de inserción en matriz 2 $atk+1..$atn" resultado=FALSE
              END CASE
         ELSE PRINTF "error de lectura en linea $i" CONTINUE resultado=FALSE
        i++
 END WHILE
return resultado

BANANER ($1){
         IF carga (1) 
             PRINTF "INGRESE EL RUN PARA CONSULTA DE CARGA ACADÉMICA")
                     read(RUN)
                     IF RUN EXISTE PRINT (carga acaddemica)
                     ELSE PRINTF("RUN NO EXISTE EN LA BASE DE DATOS")
           PRINTF "INGRESE EL SIGLA PARA CONSULTA DE LISTA DE CURSO")
                     read(sigla)
                     IF sigla EXISTE PRINT (lista de curso)
                     ELSE PRINTF("SIGLA NO EXISTE EN LA BASE DE DATOS")
}

**Comment by coniverav** [comment]
Para qué es esto? no entiendo

**Reply by TomasKetterer** [reply]
Es un pseudocódigo para ayudarles a entender un poco lo que tienen que hacer en esta entrega, no es necesario hacerlo de esta manera, pero les puede llegar a servir si se encuentran algo perdidos :)
